Source/JavaScriptCore/b3/B3LowerInt64.cpp

```diff
@@ -1184,6 +1184,41 @@ class LowerInt64 {
             setMapping(m_value, valueLo(m_value, m_index + 1), valueHi(m_value, m_index + 1));
             return;
         }
+        case AtomicStrongCAS:
+        case AtomicWeakCAS: {
+            auto atomic = m_value->as<AtomicValue>();
+            if (atomic->accessWidth() != Width64)
+                return;
+            auto expectedValue = getMapping(atomic->child(0));
+            auto newValue = getMapping(atomic->child(1));
+            auto address = atomic->child(2);
+            auto stitchedExpected = valueLoHi(expectedValue.first, expectedValue.second, m_index);
+            auto stitchedNew = valueLoHi(newValue.first, newValue.second, m_index);
+            auto stitched = insert<AtomicValue>(m_index, atomic->opcode(), m_origin, atomic->accessWidth(), stitchedExpected, stitchedNew, address);
```

Comment:
Can we copy the other fields of `AtomicValue` (specifically those of its superclass `MemoryValue`) from `m_value`? Similar to what we do for `Load` in this file.